include/RAJA/pattern/kernel/Lambda.hpp

```diff
@@ -79,6 +80,32 @@ struct StatementExecutor<statement::Lambda<LoopIndex>> {
   }
 };
 
+
+/*!
+ * A RAJA::kernel statement that invokes a lambda function
+ * with user specified arguments.
+ */
+template <camp::idx_t LoopIndex,typename... Args>
+struct StatementExecutor<statement::Lambda<LoopIndex, Args...>> {
+
+  template <typename Data>
+  static RAJA_INLINE void exec(Data &&data)
+  {
+
+    //Convert SegList, ParamList into Seg, Param types, and store in a list
+    auto targList = parser<camp::list<Args...>>::checkArgs();
+
+    //Create a tuple with the appropriate lambda arguments
+    auto argTuple = call_extractor<decltype(targList)>::make_tuple(data);
+
+    //Invoke the lambda with custom arguments
+    const int tuple_size = camp::tuple_size<decltype(argTuple)>::value;
+    invoke_lambda_with_args<LoopIndex>(std::forward<Data>(data),
```

Comment:
it would be nice if this call looked like:
`invoke_lambda_with_args<LambdaIndex, targList>(data)`

the invoke function should be able to pull out the appropriate information from the LoopData object that's passed in.

if invoke_lambda_width_args needs the make_idx_seq_t, then it should call a invoke_lambda_with_args_expanded with the appropriate index sequence... that way the caller doesn't have to know the special sauce that's needed to call it.

this would cleanup the call sites of invoke_lambda_with_args, and make it easier to maintain in the future